### PR TITLE
Fix master where old dbs have invalid PSBTs

### PR DESCRIPTION
**Pull request by rustyrussell**
After this passes CI, I'll test on my node, see if I can bring it back online!

Edit: it worked!

Changelog-None: this problam was only revealed with libwally upgrade, this release cycle.

### CODE REVIEW

File: wallet/test/run-psbt_fixup.c [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+	struct wally_psbt *psbt;
+	u8 *data;
+
+	common_setup(argv[0]);
+	chainparams = chainparams_for_network("bitcoin");
+
+	for (size_t i = 0; i < ARRAY_SIZE(psbts); i++) {
+		data = tal_hexdata(tmpctx, psbts[i], strlen(psbts[i]));
+		psbt = psbt_from_bytes(tmpctx, data, tal_bytelen(data));
+		if (psbt) {
```
Comment (instagibbs) [inline wallet/test/run-psbt_fixup.c:175]: do a quick check that both branches are taken at least once to check noop and real fixup behavior?

**Review by instagibbs — COMMENTED** [review]
LGTM

**Review by niftynei — APPROVED** [review]
ACK; nice patch for a dumb bug.

File: wallet/test/run-psbt_fixup.c (hunk 1 above)
Comment (rustyrussell) [reply wallet/test/run-psbt_fixup.c:175]: I used to have fprintf in both, but I'll re-add an assertion that 7 get fixed up.

File: wallet/psbt_fixup.c [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+
+	/* Skip global map */
+	while ((kp = fromwire_keypair(tmpctx, &cursor, &max)) != NULL);
+
+	/* Now input map */
+	changed_kp = NULL;
+	prev_cursor = cursor;
+	while ((kp = fromwire_keypair(tmpctx, &cursor, &max)) != NULL) {
+		/* PSBT_IN_PARTIAL_SIG = 0x02 */
+		if (kp->keytype == 2 && fixup_sig(kp)) {
```
[the remark this answers is not in the document]
Comment (rustyrussell) [reply wallet/psbt_fixup.c:173]: Ours do...

**Pull request merged**